### PR TITLE
chore: fix impl to api name memory leak

### PR DESCRIPTION
**Pull request by mxschmitt**
fixes https://github.com/microsoft/playwright-python/issues/1602

### CODE REVIEW

File: tests/test_reference_count_async.py [hunk 1]
```diff
@@ -44,12 +57,17 @@ async def test_memory_objects(server: Server, browser_name: str) -> None:
     gc.collect()
 
     pw_objects: defaultdict = defaultdict(int)
+    found_cdp_network_events = False
     for o in objgraph.by_type("dict"):
         name = o.get("_type")
+        if o.get("__pw__is_last_network_response_received_event", False):
+            found_cdp_network_events = True
```
Comment (mxschmitt) [inline tests/test_reference_count_async.py:64]: maybe move assert there

**Review by pavelfeldman — APPROVED** [review]

**Comment by nay-kang** [comment]
I had face this memory leak for a week. thank you for fixing it.
but I'm confusing with this code:
* there is seems no call to `from_maybe_impl` function with `visited` argument
* it has no benift to create a `visited` inside this function. in the previous version the shared `visisted` argument could cache obj between calls to speed up but will cause memory leak.

so I guess if we can remove the `visited` argument. or using an instance attribute belongs to `BrowserContext` then we can close `BrowserContext` to relase memory.

and there has another code could cause memory leak `playwright/_impl/_js_handle.py`
```python
@dataclass
class VisitorInfo:
    visited: Map[Any, int] = Map()
    last_id: int = 0

    def visit(self, obj: Any) -> int:
        assert obj not in self.visited
        self.last_id += 1
        self.visited[obj] = self.last_id
        return self.last_id
```
the `visited` attribute in the `VisitorInfo` is a class attribute, which will share between instance and will not release memory until reload the module.

all this code are belongs to the same commit:`ed13a53281bc213e1a4341c50d47fbc65670288f`

**Comment by mxschmitt** [comment]
I see one, see [here](https://github.com/microsoft/playwright-python/blob/8ecd961bdb532dc658151f1fa2c9ecc9b062182c/playwright/_impl/_impl_to_api_mapping.py#L55).

Will create a follow-up with this potential VisitorInfo leak, thanks for pointing this out.

**Comment by nay-kang** [comment]
hi @mxschmitt ,you memtioned the `from_maybe_impl` function with `visited` argument call [here](https://github.com/microsoft/playwright-python/blob/8ecd961bdb532dc658151f1fa2c9ecc9b062182c/playwright/_impl/_impl_to_api_mapping.py#L55) 
```python
o[name] = self.from_maybe_impl(value, visited)
```  
 is a recursive call to himself, so I think it could run well without `visisted` argument

**Pull request merged**